srt-tokio/src/builder.rs

```diff
@@ -238,9 +257,21 @@ impl SrtSocketBuilder {
 
     /// Connects to the remote socket. Resolves when it has been connected successfully.
     pub async fn connect(self) -> Result<SrtSocket, io::Error> {
-        let la = self.local_addr;
+        let addr = match self.conn_type {
+            ConnInitMethod::Connect(addr, _) => addr.ip(),
+            ConnInitMethod::Rendezvous(addr) => addr.ip(),
+            _ => Ipv4Addr::UNSPECIFIED.into(),
```

Comment:
I think it's a bit cleaner to put the explicit variant, which would also make sure this doesn't compile if more modes are added